### PR TITLE
Extend the chart to support to create the ingress resource. #5

### PR DESCRIPTION
**Pull request by cmoulliard**
## What 
- Extend the existing helm chart to support to create the ingress resource. 
- Add a new ingress template
- Add within the values.yaml files the new ingress parameter
See issue #5

## How to
- Create a backstage project using command `npx @backstage/create-app`
- Add the missing package and build/push the image
```
yarn add --cwd packages/backend better-sqlite3
yarn build
yarn build-image -t backstage:dev
kind load docker-image backstage:dev
```
- Create a configmap
```
DOMAIN_NAME="192.168.1.90.nip.io"
cat <<EOF > $HOME/backstage/app-config.extra.yaml
app:
  baseUrl: http://backstage.$DOMAIN_NAME
  title: Backstage
backend:
  baseUrl: http://backstage.$DOMAIN_NAME
  cors:
    origin: http://backstage.$DOMAIN_NAME

  # This is for local development only, it is not recommended to use this in production
  # The production database configuration is stored in app-config.production.yaml
  database:
    client: better-sqlite3
    connection: ':memory:'
  cache:
    store: memory
techdocs:
  builder: 'local' # Alternatives - 'external'
  generator:
    runIn: 'docker' # Alternatives - 'local'
  publisher:
    type: 'local' # Alternatives - 'googleGcs' or 'awsS3'. Read documentation for using alternatives.    
EOF
```
- Create the values.yaml fileto be used to install the helm chart
```
cat <<EOF > $HOME/backstage/my-values.yml
ingress:
  enabled: true
  host: backstage.$DOMAIN_NAME
backstage:
  image:
    registry: "docker.io/library"
    repository: "backstage"
    tag: "dev"
  extraAppConfig:
    - filename: app-config.extra.yaml
      configMapRef: my-app-config      
EOF
```
- deploy the helm chart and configmap
```
helm install -f $HOME/backstage/my-values.yml --create-namespace -n backstage my-backstage ./charts/backstage
kubectl create configmap my-app-config -n backstage --from-file=app-config.extra.yaml=$HOME/backstage/app-config.extra.yaml
```

### CODE REVIEW

**Review by vinzscam — COMMENTED** [review]
This is great @cmoulliard! 🙏 🚀

File: charts/backstage/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{{- if .Values.ingress.enabled }}
+apiVersion: networking.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (cmoulliard) [reply charts/backstage/templates/ingress.yaml:2]: I like the idea to use the bitnami `common.capabilities.ingress.apiVersion` function to return the appropriate apiVersion for ingress. 

**Remark**:  if we do that, then additional logic will be needed to support the old and new syntax for the [`Backend`](https://github.com/bitnami/charts/tree/master/bitnami/common#ingress) and `PathType` sections at least.

**Review by vinzscam — APPROVED** [review]

File: charts/backstage/templates/ingress.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vinzscam) [reply charts/backstage/templates/ingress.yaml:2]: sounds good. We can do it later if you prefer.

And if you would like to release a new version of the chart that includes your changes, feel free to manually bump [this value](https://github.com/vinzscam/backstage-chart/blob/main/charts/backstage/Chart.yaml#L18) in this PR.

[the remark this answers is not in the document]
Comment (cmoulliard) [reply charts/backstage/templates/ingress.yaml:2]: I will then create another ticket to take care about my remark/proposition to handle old/new syntax for backend and PathType

**Comment by cmoulliard** [comment]
Can you approve it please to merge the code ? @vinzscam

**Pull request merged**